app/services/check_decision_tree.rb

```diff
@@ -22,4 +24,13 @@ def after_under_age
       edit('/steps/conviction/conviction_date')
     end
   end
+
+  def after_remove_check
+    if disclosure_check.disclosure_report.disclosure_checks.any?
```

Comment:
Probably best to change to `disclosure_check.disclosure_report.disclosure_checks.completed.any?` (note the _completed_ scope) to ensure we only return to the CYA page if there are any other completed cautions or convictions otherwise in the case of kick out pages etc. there would be left overs and the user will go back to an empty CYA.